osbenchmark/synthetic_data_generator/models.py

```diff
@@ -0,0 +1,129 @@
+# GitHub history for details.
+
+import os
+from typing import Optional, Dict, List, Any, Union
+import re
+
+from pydantic import BaseModel, Field, field_validator
+
+GB_TO_BYTES = 1024 ** 3
+
+class SettingsConfig(BaseModel):
+    workers: Optional[int] = Field(default_factory=os.cpu_count)
+    max_file_size_gb: Optional[int] = 40
+    docs_per_chunk: Optional[int] = 10000
+
+    @field_validator('workers', 'max_file_size_gb', 'docs_per_chunk')
+    def validate_values_are_positive_integers(cls, v):
+        if v is not None and v <= 0:
+            raise ValueError('Value in Settings portion must be a positive integer.')
+
+        return v
+
+class CustomGenerationValuesConfig(BaseModel):
+    custom_lists: Optional[Dict[str, List[Any]]] = None
+    custom_providers: Optional[List[Any]] = None
+
+    @field_validator('custom_lists')
+    def validate_custom_lists(cls, v):
+        if v is not None:
+            for key, value in v.items():
+                if not isinstance(key, str):
+                    raise ValueError('All keys within custom_lists of CustomGenerationValues section must be strings')
+                if not isinstance(value, list):
+                    raise ValueError(f"Value for key '{key}' must be a list.")
+        return v
+
+class GeneratorParams(BaseModel):
+    # Integer / Long Params
+    min: Optional[Union[int, float]] = None
+    max: Optional[Union[int, float]] = None
+
+    # Date Params
+    start_date: Optional[str] = None
+    end_date: Optional[str] = None
+    format: Optional[str] = None
+
+    # Text / Keywords Params
+    must_include: Optional[List[str]] = None
+    choices: Optional[List[str]] = None
+
+    class Config:
+        extra = 'forbid'
+
+class FieldOverride(BaseModel):
+    generator: str
+    params: GeneratorParams
+
+    @field_validator('generator')
+    def validate_generator_name(cls, v):
+        valid_generators = [
+            'generate_text',
+            'generate_keyword',
+            'generate_integer',
+            'generate_long',
+            'generate_short',
+            'generate_byte',
+            'generate_float',
+            'generate_double',
+            'generate_boolean',
+            'genearte_date',
+            'generate_ip',
+            'generate_geopoint',
+            'generate_object',
+            'generate_nested'
+        ]
+
+        if v not in valid_generators:
+            raise ValueError(f"Generator '{v}' mentioned in FieldOverrides not among valid generators: {valid_generators}")
+        return v
+
+class MappingGenerationValuesConfig(BaseModel):
+    generator_overrides: Optional[Dict[str, GeneratorParams]] = None
+    field_overrides: Optional[Dict[str, FieldOverride]] = None
+
+    @field_validator('generator_overrides')
+    def validate_generator_types(cls, v):
+        if v is not None:
+            valid_generator_types = ['integer', 'long', 'float', 'double', 'date', 'text', 'keyword']
```

Comment:
Should `short`,`byte`, etc. be in here too?

Reply:
Good call out. I plan to open a subsequent PR for adding more complications to Mapping SDG but will update this here. I'll add all except for object.